src/cnaas_nms/db/job.py

```diff
@@ -0,0 +1,120 @@
+import enum
+import datetime
+import json
+from typing import Optional
+
+from sqlalchemy import Column, Integer, Unicode
+from sqlalchemy import Enum, DateTime
+from sqlalchemy import ForeignKey
+from sqlalchemy.dialects.postgresql.json import JSONB
+from sqlalchemy.orm import relationship
+from nornir.core.task import AggregatedResult
+
+import cnaas_nms.db.base
+import cnaas_nms.db.device
+from cnaas_nms.confpush.nornir_helper import nr_result_serialize, NornirJobResult
+from cnaas_nms.scheduler.jobresult import StrJobResult, DictJobResult
+from cnaas_nms.db.helper import json_dumper
+from cnaas_nms.tools.log import get_logger
+
+
+logger = get_logger()
+
+
+class JobStatus(enum.Enum):
+    UNKNOWN = 0
+    SCHEDULED = 1
+    RUNNING = 2
+    FINISHED = 3
+    EXCEPTION = 4
+
+    @classmethod
+    def has_value(cls, value):
+        return any(value == item.value for item in cls)
+
+    @classmethod
+    def has_name(cls, value):
+        return any(value == item.name for item in cls)
+
+
+class Job(cnaas_nms.db.base.Base):
+    __tablename__ = 'job'
+    __table_args__ = (
+        None,
+    )
+    id = Column(Integer, autoincrement=True, primary_key=True)
+    status = Column(Enum(JobStatus), default=JobStatus.SCHEDULED)
+    scheduled_time = Column(DateTime, default=datetime.datetime.utcnow)
+    start_time = Column(DateTime)
+    finish_time = Column(DateTime)
+    function_name = Column(Unicode(255))
+    scheduled_by = Column(Unicode(255))
+    comment = Column(Unicode(255))
+    ticket_ref = Column(Unicode(32))
```

Comment:
Are these columns used or are they added for future use?

Reply:
These are for future use currently, ticket_ref is a field to enter a reference for a ticket in some external ticketing system